### PR TITLE
ToolTip: IsOpen, Placement, Offset

### LINKED ISSUE #962: Allow showing ToolTips on demand
Not just on pointer entering (e.g. add an `IsOpen` property, like WPF)

### PR DESCRIPTION
**Pull request by aelij**
Fixes #962

### CODE REVIEW

File: samples/ControlCatalog/Pages/ToolTipPage.xaml [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+                </ToolTip.Tip>
+                <TextBlock>Hover Here</TextBlock>
+            </Border>
+            <Border Background="{StyleResource ThemeAccentBrush}"
+                    Margin="5"
+                    Padding="50"
+                    ToolTip.Tip="Another tip">
+                <TextBlock>And Here</TextBlock>
+            </Border>
+        </StackPanel>
```
Comment (grokys) [inline samples/ControlCatalog/Pages/ToolTipPage.xaml]: As a demo I find this to be quite confusing. Is there a better way of demonstrating this?

File: src/Avalonia.Controls/ToolTipService.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
+using System;
+using Avalonia.Input;
+using Avalonia.Threading;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Handeles <see cref="ToolTip"/> interaction with controls.
+    /// </summary>
+    internal sealed class ToolTipService
```
Comment (grokys) [inline src/Avalonia.Controls/ToolTipService.cs]: Why did you move these methods out of the `ToolTip` class itself? Just to separate concerns? I don't have a problem with that, just wondering. Also considering the fact that this class is `internal` and seems to be a singleton, why not make it a static class?

**Review by grokys — APPROVED** [review]
Looks good! I find the demo a bit confusing, but I'm not going to block the PR based on that.

File: src/Avalonia.Controls/ToolTipService.cs (hunk 2 above)
Comment (aelij) [reply src/Avalonia.Controls/ToolTipService.cs]: Yes, separate concerns. I wanted to encapsulate all the control interaction logic; it may become even more complex in the future. Regarding `static`, I don't like having static state (the timer in this case). What if wanted to support multiple Dispatchers in the future? This class would have to be instantiated per UI thread. It's also more easily testable.

File: samples/ControlCatalog/Pages/ToolTipPage.xaml (hunk 1 above)
Comment (aelij) [reply samples/ControlCatalog/Pages/ToolTipPage.xaml]: I'm open to suggestions. I wanted to have two tool tips, one relative to the target, that can be triggered by the check box, and another that's relative to the mouse. Also wanted to check how well moving between two tool tips works.

File: src/Avalonia.Controls/ToolTipService.cs (hunk 2 above)
Comment (grokys) [reply src/Avalonia.Controls/ToolTipService.cs]: 👍 makes sense

File: samples/ControlCatalog/Pages/ToolTipPage.xaml (hunk 1 above)
Comment (grokys) [reply samples/ControlCatalog/Pages/ToolTipPage.xaml]: Not sure, but how about this?

- Have the default behavior border/tooltip first
- Put the "Show ToolTip" checkbox _below_ the (now second) border and change its text to "ToolTip Open"

That might be clearer, what do you think?

Comment (aelij) [reply samples/ControlCatalog/Pages/ToolTipPage.xaml]: Sure, makes sense

File: src/Avalonia.Controls/ToolTipService.cs (hunk 2 above)
Comment (kekekeks) [reply src/Avalonia.Controls/ToolTipService.cs]: >why not make it a static class

Becaise static variables are *evil*. We even have an issue about that - https://github.com/AvaloniaUI/Avalonia/issues/1013
